Tests/Tests/STPFormEncoderTest.m

```diff
@@ -98,6 +98,21 @@ - (void)testFormEncoding_arrayValue {
     XCTAssertEqualObjects([self encodeObject:testObject], @"test_object[test_array_property][]=1&test_object[test_array_property][]=2&test_object[test_array_property][]=3&test_object[test_property]=success");
 }
 
+- (void)testFormEncoding_arrayOfEncodable {
+    STPTestFormEncodableObject *testObject = [STPTestFormEncodableObject new];
+
+    STPTestFormEncodableObject *inner1 = [STPTestFormEncodableObject new];
+    inner1.testProperty = @"inner1";
+    STPTestFormEncodableObject *inner2 = [STPTestFormEncodableObject new];
+    inner2.testArrayProperty = @[@"inner2"];
+
+    testObject.testArrayProperty = @[inner1, inner2];
+
+    XCTAssertEqualObjects([self encodeObject:testObject],
```

Comment:
This fails, `encodeObject:` returns:

```
 "test_object[test_array_property][]=<STPTestFormEncodableObject: 0x7fb845d0ffd0>"
"&test_object[test_array_property][]=<STPTestFormEncodableObject: 0x7fb845d10010>"
```

Reply:
As far as I can tell, Android would serialize this as

```java
 "test_object[test_array_property][][test_property]=inner1" +
"&test_object[test_array_property][][test_array_property][]=inner2"
```

Which could either be an array with a single object `[{ test_property=inner1, test_array_property=[inner2] }]` **or** it could be an array with two objects `[{ test_property=inner1 }, { test_array_property=[inner2] }]`